File: 0237-delete-node-in-a-linked-list/0237-delete-node-in-a-linked-list.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode(int x) : val(x), next(NULL) {}
 * };
 */
class Solution {
public:
    void deleteNode(ListNode* node) {
        ListNode* temp = node ;
        while(temp){
            node = node-> next;
            temp -> val = node -> val ;
            temp -> next = node -> next ;
            temp = temp -> next ;
        }

    }
};

File: 0237-delete-node-in-a-linked-list/0237-delete-node-in-a-linked-list.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode(int x) { val = x; }
 * }
 */
class Solution {
    public void deleteNode(ListNode node) {
        ListNode temp = node ;
        node.val = node.next.val;   // Copy the value from the next node
        node.next = node.next.next; 
        
    }
}